hazelcast/src/test/java/com/hazelcast/map/impl/EventListenerCounterTest.java

```diff
@@ -0,0 +1,111 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.NearCacheConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hazelcast.test.Accessors.getNodeEngineImpl;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class EventListenerCounterTest extends HazelcastTestSupport {
+
+    EventListenerCounter eventListenerCounter = new EventListenerCounter();
+
+    @Test
+    public void testGetOrCreatePerMapEventListenerCounter() {
+        AtomicInteger counter1 = eventListenerCounter.getOrCreateCounter("A");
+        AtomicInteger counter2 = eventListenerCounter.getOrCreateCounter("A");
+
+        assertTrue(counter1 == counter2);
+    }
+
+    @Test
+    public void testRemoveFromPerMapEventListenerCounter() {
+        AtomicInteger counter1 = eventListenerCounter.getOrCreateCounter("B");
+        eventListenerCounter.removeCounter("B", counter1);
+
+        AtomicInteger counter2 = eventListenerCounter.getOrCreateCounter("B");
+
+        assertTrue(counter1 != counter2);
+    }
+
+    @Test
+    public void testIncPerMapEventListenerCounter() {
+        int countBefore = eventListenerCounter.getOrCreateCounter("A").get();
+        eventListenerCounter.incCounter("A");
+        int countAfter = eventListenerCounter.getOrCreateCounter("A").get();
+
+        assertEquals(countBefore + 1, countAfter);
+    }
+
+    @Test
+    public void testDecPerMapEventListenerCounter() {
+        eventListenerCounter.incCounter("A");
+        int countBefore = eventListenerCounter.getOrCreateCounter("A").get();
+
+        eventListenerCounter.decCounter("A");
+        int countAfter = eventListenerCounter.getOrCreateCounter("A").get();
+
+        assertEquals(countBefore - 1, countAfter);
+    }
+
+    @Test
+    public void near_cache_invalidation_listener_does_not_cause_map_container_creation_on_join() {
+        Config config = new Config();
```

Comment:
maybe use `smallInstanceConfig()` to avoid excessive thread creation?

Reply:
fixed.